tool/site/client-react/src/pages/myPrescriptionsPage.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react"
+import {useContext} from "react"
+import {Label, Table} from "nhsuk-react-components"
+import {AppContext} from "../index"
+import LongRunningTask from "../components/longRunningTask"
+import {axiosInstance} from "../requests/axiosInstance"
+import PrescriptionActions from "../components/prescriptionActions"
+
+const MyPrescriptionsPage: React.FC = () => {
+  const {baseUrl} = useContext(AppContext)
+  const retrievePrescriptionsTask = () => retrievePrescriptions(baseUrl)
+
+  return (
+    <LongRunningTask<Prescriptions> task={retrievePrescriptionsTask} loadingMessage="Retrieving prescriptions.">
+      {prescriptions => {
+        if (!prescriptions.any) {
+          return (
+            <>
+              <Label isPageHeading>My Prescriptions</Label>
+              <p>You do not have any active prescriptions. Once you create or release a prescription you will see it here</p>
+            </>
+          )
+        }
+        return (
+          <>
+            <Label isPageHeading>My Prescriptions</Label>
+            <PrescriptionGroupTable
+              name="Sent Prescriptions"
+              description="Prescriptions ready to release"
+              prescriptions={prescriptions.sentPrescriptions}
+              actions={{view: true, release: true}}
+            />
+            <PrescriptionGroupTable
+              name="Released Prescriptions"
+              description="Prescriptions ready to dispense"
+              prescriptions={prescriptions.releasedPrescriptions}
+              actions={{view: true, dispense: true}}
+            />
+          </>
+        )
+      }}
+    </LongRunningTask>
+  )
+}
+
+interface Prescriptions {
+  any: Boolean
+  sentPrescriptions: Array<PrescriptionSummary>
+  releasedPrescriptions: Array<PrescriptionSummary>
+}
+
+interface PrescriptionSummary {
+  id: string
+}
+
+async function retrievePrescriptions(baseUrl: string): Promise<Prescriptions> {
+  return await (await axiosInstance.get<Prescriptions>(`${baseUrl}prescriptions`)).data
+}
+
+interface PrescriptionGroupTableProps {
+  name: string
+  description: string
+  prescriptions: Array<{id: string}>
+  actions: PrescriptionActionProps
+}
+
+interface PrescriptionActionProps {
+  view?: boolean
+  release?: boolean
+  dispense?: boolean
+  claim?: boolean
+}
+
+export const PrescriptionGroupTable: React.FC<PrescriptionGroupTableProps> = ({
```

Comment:
wonder if its worth separating this out into its own file?

Reply:
would be good to decouple the tests for this component from the `MyPrescriptionsPage` component

Reply:
have split out the components from the page